app/src/main/java/de/grobox/transportr/settings/SettingsFragment.kt

```diff
@@ -91,6 +102,34 @@ class SettingsFragment : PreferenceFragmentCompat() {
                 true
             }
         }
+
+        arrayOf(PROXY_ENABLE, PROXY_HOST, PROXY_PORT, PROXY_PROTOCOL).forEach { prefKey ->
+            (findPreference(prefKey) as Preference?)?.let { pref ->
+                pref.setOnPreferenceChangeListener { _, _ ->
+                    updateGlobalHttpProxy(settingsManager.proxy, manager)
+                    true
+                }
+            }
+        }
+
+        findPreference<ValidatedEditTextPreference>(PROXY_HOST)!!.validate = { value ->
+            /*
+            FIXME: These validation functions are marked unstable, maybe find a replacement
+
+             Valid should be everything accepted by InetAddress.getByName(...),
+             that is: IPv4 addresses, IPv6 addresses, hostnames and FQDNs.
+
+             Additionally, no network requests should be performed as part of
+             the validation, so simply using InetAddress.getByName(...) with a
+             try/catch is out of the question since it might try to resolve
+             domain names.
+             */
+            InternetDomainName.isValid(value) || HostSpecifier.isValid(value)
```

Comment:
Couldn't we use something like [`Patterns.DOMAIN_NAME`](https://developer.android.com/reference/android/util/Patterns#DOMAIN_NAME) and `Patterns.IP_ADDRESS`? For some reason the current check accepts "g" as host for me which doesn't make too much sense I'd guess

Reply:
Thanks, I didn't know about those. Although "g" should theoretically be okay since it's a valid hostname and it could theoretically also resolve. This is also how localhost works. (And "localhost" is also not accepted with Patterns.)